openmc/filter.py

```diff
@@ -964,6 +965,127 @@ def get_pandas_dataframe(self, data_size, stride, **kwargs):
         return pd.concat([df, pd.DataFrame(filter_dict)])
 
 
+class CollisionFilter(Filter):
+    """Bins tally events based on the number of collisions.
+
+    Parameters
+    ----------
+    values : Iterable of Int
+        A list or iterable of the number of collisions, as integer values. 
+        The events whose post-scattering collision number equals one of 
+        the provided values will be counted. 
+    filter_id : int
+        Unique identifier for the filter
+
+    Attributes
+    ----------
+    values : numpy.ndarray
+        An array of integer values representing the number of collisions events
+        by which to filter
```

Comment:
For this filter, it doesn't really make sense to have a separate `values` attribute. I would just keep bins. For some filters, `values` represents something separate from `bins` (see EnergyFilter for example).

Reply:
I used EnergyFilter as an example of how filters are typically written in OpenMC, so I kept bins separate from values in case there was some distinction use that I had not picked up on. Glad to know it's nothing like that. I'll make the change, thanks!